libtiledbvcf/src/stats/allele_count.cc

```diff
@@ -465,39 +457,40 @@ void AlleleCount::process(
   // Sort ALT alleles and normalize GT
   //  - haploid GT = 1
   //  - diploid GT = 0,1 or 1,1 or 1,2
-  std::string alt;
-  std::string gt;
-  if (ngt == 1) {
-    // haploid
-    alt = rec->d.allele[gt0];
-    gt = "1";
+  std::string sulected_alt;
+  std::string selected_gt;
+  // diploid
+  std::vector<std::string_view> alt(ngt);
+  for (int i = 0; i < ngt; i++) {
+    alt[i] = gt_missing[i] ? "." : rec->d.allele[gt[i]];
+  }
+
+  if (gt_missing[0] || gt_missing[1]) {
+    selected_gt = ".,1";
+    sulected_alt = gt_missing[0] ? alt[1] : alt[0];
+  } else if (!gt[0] || !gt[1]) {
+    selected_gt = "0,1";
+    sulected_alt = gt[0] ? alt[0] : alt[1];
+  } else if (gt[0] == gt[1]) {
+    selected_gt = "1,1";
+    sulected_alt = alt[0];
   } else {
-    // diploid
-    std::string_view alt0{gt0_missing ? "." : rec->d.allele[gt0]};
-    std::string_view alt1{gt1_missing ? "." : rec->d.allele[gt1]};
-
-    if (gt0_missing || gt1_missing) {
-      gt = ".,1";
-      alt = gt0_missing ? alt1 : alt0;
-    } else if (!gt0 || !gt1) {
-      gt = "0,1";
-      alt = gt0 ? alt0 : alt1;
-    } else if (gt0 == gt1) {
-      gt = "1,1";
-      alt = alt0;
+    selected_gt = "1,2";
```

Comment:
We haven't defined how ployploid variants should be represented in the allele count array. We could release the variant stats and IAF changes in this PR and follow up with the allele count changes in another PR.

Reply:
There's no logic yet to handle polyploid variants, just a change from a scalar to vector representation in anticipation of the next step.